opsdroid/core.py

```diff
@@ -97,54 +107,95 @@ def critical(self, error, code):
         _LOGGER.critical(error)
         self.exit()
 
-    def call_stop(self):
-        """Signal handler to call disconnect and stop."""
-        future = asyncio.ensure_future(self.disconnect())
-        future.add_done_callback(self.stop)
-        return future
+    @staticmethod
+    def handle_async_exception(loop, context):
+        """Handle exceptions from async coroutines."""
+        _LOGGER.error("Caught exception")
+        _LOGGER.error(context)
+
+    def is_running(self):
+        """Check whether opsdroid is running."""
+        return self._running
+
+    async def handle_signal(self):
+        """Handle signals."""
+        self._running = False
+        await self.unload()
+
+    def run(self):
+        """Start the event loop."""
+        _LOGGER.info(_("Opsdroid is now running, press ctrl+c to exit."))
+        if not self.is_running():
+            self._running = True
+            while self.is_running():
+                pending = asyncio.Task.all_tasks()
+                try:
+                    self.eventloop.run_until_complete(asyncio.gather(*pending))
+                except asyncio.CancelledError:
```

Comment:
Could this be replaced with `with contextlib.suppress(asyncio.CancelledError)`?

Reply:
Probably :)